doc/chapters/gateways/index.rst

```diff
@@ -352,82 +352,35 @@ The ID used for the Network gateway is: ``network``
-``ACCEPT`` and ``DROP``.
-
-As mentioned above, different order of the rules can have profound different meaning. Following are
-an example attempting to reject all traffic from example.com on ports 1234 to 5678 and adding an
-exception for port 1423.::
+or ``INCOMING`` network traffic. The validation is done by filtering network traffic on ``host``, ``ports`` and 
+``protocols``, where ``host`` is either hostname or ip address of a destination or source depending on context,
+``ports`` specifies which ports to filter on and ``protocols`` specifies which protocols to filter. 
 
-    [
-        {
-            "type": "INCOMING",
-            "priority": 1,
-            "rules": [
-                         { "host": "example.com", "port": "1234:5678", "target": "REJECT"},
-                         { "host": "example.com", "port": 1423, "target": "ACCEPT"},
-                     ],
-            "default": "DROP"
-        }
-    ]
-
-However, since the rejection of the ports 1234 to 5678 is declared before the exception to the port
-1423, which is included in the range of ports rejected, the exception will not have any effect. In
-order to achive this the order of the rules has to be changed::
-
-    [
-        {
-            "type": "INCOMING",
-            "priority": 1,
-            "rules": [
-                         { "host": "example.com", "port": 1423, "target": "ACCEPT"},
-                         { "host": "example.com", "port": "1234:5678", "target": "REJECT"},
-                     ],
-            "default": "DROP"
-        }
-    ]
-
-Now, any incoming traffic from example.com on port 1423 will be accepted and traffic on the rest
-of the range will be rejected.
-
-
-In order to not make the listing of capabilities, e.g. in application manifests, order dependent,
-``priority`` is used when merging network gateway configurations of the same type. This is specified
-as an unsigned int > 0 where 1 describes the highest priority.
+``direction`` and ``allow`` list are mandatory for Network Gateway configuration. There are only two valid values for 
+``direction``: ``INCOMING`` and ``OUTGOING``. In each entry in ``allow`` list only ``host`` is mandatory. ``host`` 
+can be specific hostname or ip address and also ``*`` indicating all available ip sources. When ``ports`` is not specified, 
+the rule applies to all ports.  ``ports`` are valid between 0 and 65536. There are threee valid values for ``protocols``: 
+``"tcp"``, ``"udp"`` and ``"icmp"``. When ``protocols`` is not specified in the ``allow`` list item, the rule applies to 
+all protocols. When there is an item which has ports without protocols, tcp protocol will be applied to the rule. 
+NetworkGateway is programmed to drop all packages that does not match any entry in ``allow`` list.
 
-An example of valid network gateway configuration::
+Following is an example attempting to reject all ping requests except example.com. Only port 53 on tcp and udp protocols 
```

Comment:
I think it would be better to write "The following is an example which rejects all ping requests..." (without "attempts to")